RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -132,12 +132,35 @@ class PumpOpsSynchronous {
             throw lastError
         }
     }
+    
+    private func pumpResponding() -> Bool {
+        do {
+            let msg = makePumpMessage(.GetPumpModel)
+            let response = try sendAndListen(msg, retryCount: 1)
+            
+            if response.messageType == .GetPumpModel && (response.messageBody as? GetPumpModelCarelinkMessageBody) != nil {
+                return true
+            }
+        } catch {
+        }
+        return false
+    }
+    
+
 
     private func wakeup(duration: NSTimeInterval = NSTimeInterval(minutes: 1)) throws {
         guard !pump.isAwake else {
             return
         }
         
+        if pumpResponding() {
+            NSLog("Pump responding despite our wake timer having expired. Extending timer")
+            // By my observations, the pump stays awake > 1 minute past last comms. Usually
```

Comment:
👍👍👍
On Sun, Sep 11, 2016 at 8:41 PM Pete Schwamb <notifications@github.com>
wrote:

> In RileyLinkKit/PumpOpsSynchronous.swift
> <https://github.com/ps2/rileylink_ios/pull/227#discussion_r78312541>:
>
> >
> >      private func wakeup(duration: NSTimeInterval = NSTimeInterval(minutes: 1)) throws {
> >          guard !pump.isAwake else {
> >              return
> >          }
> >
> > +        if pumpResponding() {
> > +            NSLog("Pump responding despite our wake timer having expired. Extending timer")
> > +            // By my observations, the pump stays awake > 1 minute past last comms. Usually
>
> Just tested again on a 523. Waited for the app to wake it up, then waited
> 40s and did a fetch model, then turned off the RL to ensure no more comms.
> The pump continued to use power (which is indicative of RX) for 1m45s.
>
> The "Pump responding despite our wake timer having expired. Extending
> timer" message also is seen in the logs fairly frequently on the 523 while
> I've been playing around.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ps2/rileylink_ios/pull/227/files/81afe8f84ff79d8e719a926421c8ca2a7ad2891e#r78312541>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAM_jId072Kqnk76LKg9z3lfJkSFpsyRks5qpMn5gaJpZM4J5ZK6>
> .
>